proxy/stats_handler.go

```diff
@@ -0,0 +1,215 @@
+package proxy
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/gorilla/websocket"
+
+	"github.com/rancherio/websocket-proxy/common"
+)
+
+type StatsHandler struct {
+	backend         backendProxy
+	parsedPublicKey interface{}
+}
+
+func (h *StatsHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	multiHost := false
+
+	if strings.HasSuffix(req.URL.Path, "project/") || strings.HasSuffix(req.URL.Path, "service/") {
+		multiHost = true
+	}
+
+	hostKeys, urls, authed := h.auth(req, multiHost)
+	if !authed {
+		return
+	}
+
+	upgrader := websocket.Upgrader{
+		CheckOrigin: func(r *http.Request) bool { return true },
+	}
+	ws, err := upgrader.Upgrade(rw, req, nil)
+	if err != nil {
+		http.Error(rw, "Failed to upgrade connection.", 500)
+		return
+	}
+
+	msgKeys := make([]string, len(hostKeys))
+	respChannels := make([]<-chan common.Message, len(hostKeys))
```

Comment:
This function feels a little messy with separate arrays for msgKeys, respChannels, hostKeys, and urls.

@wlan0 do you think it would be cleaner if we had a single data structure that house a msgkey, respChannel, hostKey, and url for each backend connection? Then we would have just a single array and perhaps instead of passing pos into the "Send response messages to client" go routine, we could just pass the structure with all the necessary data.